README.md

```diff
@@ -2,81 +2,42 @@ This is the code and configuration to carry out the antivirus checks on a single
+The images are tagged with the jenkins build number and then tagged with the stage. This allows us to have different sets of dependencies for different stages. 
 
-`mkdir lambda`
+### Jenkinsfile-test 
+This runs git secrets and runs the python tests. This is the standard multibranch pipeline job which runs on PRs and merge to master. If this runs on the master branch, it will trigger the bundle job.
 
-Copy the rules file into the lambda directory
+### Jenkinsfile-bundle
+This creates the lambda zip using the stored docker images that were built using the Dockerfile-build job and the latest python code from the project. The zip is uploaded to s3 and the deploy job is triggered.  
 
-`docker cp rules:/rules/output ./lambda`
+### Jenkinsfile-deploy
+This updates the lambda with the zip file from S3.
 
-Unzip the dependencies into the lambda directory
+## Deploying changes
 
-`unzip -q dependencies.zip -d ./lambda`
+There are two situations where the lambda will need to be redeployed.
 
-Copy the matcher file to the lambda directory
+### Docker image updates
+If there are vulnerabilities in the current docker images and they need to be rebuilt or dependencies need to be updated, but there are no changes to the Jenkinsfile-build or the Dockerfiles themselves, then run the [build](https://jenkins.tdr-management.nationalarchives.gov.uk/job/TDR%20Antivirus%20Build/) job. This will rebuild all the images with the latest base images and the latest dependencies from the package managers.
 
-`cp src/matcher.py ./lambda`
+If there are changes to the Dockerfile or Jenkinsfile-build, this will need to be reviewed and merged to master as normal, then the build job will need to be run again. 
 
-Enter the lambda directory
+The job will tag the images with the build number and the stage provided in the parameters. It will then run the bundle and deploy jobs to deploy the changes to the lambda.
 
-`cd lambda`
-
-Zip the contents
-
-`zip -r9 function.zip .`
-
-Upload to an S3 bucket
-
-`aws s3 cp function.zip s3://tdr-backend-code-mgmt/yara-av.zip`
-
-Update the lambda function in the environment account. You will need credentials for whichever environment you're deploying to
-
-`aws lambda update-function-code --function-name tdr-yara-av-$STAGE --s3-bucket tdr-backend-checks-$STAGE --s3-key yara-av.zip`
-
-Publish a new lambda version. This is how we keep track of the database version
-
-`aws lambda publish-version --function-name tdr-yara-av-$STAGE`
-
-Clean up
-
-```
-cd ..
-rm -rf lambda dependencies.zip
-docker stop rules dependencies
-```
+### Python code updates
+This includes our own custom code and any dependencies in requirements.txt This shouldn't need any manual intervention as it should trigger the [test](https://jenkins.tdr-management.nationalarchives.gov.uk/job/TDR%20Antivirus%20Test/) job which in turn runs the bundle and deploy jobs. You can run the test job manually if you need to for any reason. 
```

Comment:
Does this only apply to integration? How do you deploy changes to  staging?

Reply:
It's just a case of running the deploy job, I've added in a line to say this.